src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -307,10 +322,17 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
 }
 
 object NIHWhitelistUtils extends FireCloudRequestBuilding {
-  def getAllUserValuesForKey(key: String)(implicit arf: ActorRefFactory, ec: ExecutionContext): Future[Map[String, String]] = {
+  def getAllUserValuesForKey(key: String, userId: Option[String] = None)(implicit arf: ActorRefFactory, ec: ExecutionContext): Future[Map[String, String]] = {
     val pipeline = addAdminCredentials ~> sendReceive
+
+    val queryParams = userId match {
+      case Some(sub) => Map("key"->key, "userId"->sub)
+      case None      => Map("key"->key)
+    }
+    val queryUri = Uri(UserService.remoteGetQueryURL).withQuery(queryParams)
```

Comment:
`remoteGetQueryURL` currently ends in a `?`.  I assume this is no longer necessary with `withQuery`, but is it harmful?  

Consider removing the `?` from `userprofile.getQuery` in `application.conf`